File: backend/src/main/kotlin/ru/itmo/db/coursework/api/dto/MachineDto.kt
package ru.itmo.db.coursework.api.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class MachineDto(
        var id: Int,
        var name: String,
        @get:JsonProperty("size_x")
        var sizeX: Int,
        @get:JsonProperty("size_y")
        var sizeY: Int,
        var price: Long,
        var recipes: Set<MachineRecipeDto>
)

File: frontend/src/api/dto/MachineDto.ts
import MachineRecipeDto from "$src/api/dto/MachineRecipeDto";

export default class MachineDto {
  constructor(
    public id: number,
    public name: string,
    public size_x: number,
    public size_y: number,
    public price: number,
    public recipes: MachineRecipeDto[]
  ) {}
}